app/controllers/restaurants_controller.rb

```diff
@@ -3,6 +3,11 @@ class RestaurantsController < ApplicationController
 
   def index
     @restaurants = Restaurant.all
```

Comment:
You don't need this line since you are setting @restaurants inside the if/else below